### PR TITLE
Upstream changes

### PR DESCRIPTION
**Pull request by amilcarlucas**
This will bring us mostly in sync with upstream mavlink/mavlink repository
WIP commits have not been ported

### CODE REVIEW

**Comment by amilcarlucas** [comment]
I will do the ardupilot code changes once this get merged.

File: message_definitions/v1.0/common.xml [hunk 1]
```diff
@@ -4295,12 +4638,70 @@
...
+      <field type="uint8_t[30]" name="sensor" enum="MAV_SENSOR_TYPE">An array describing the sensor associated with the field (see MAV_SENSOR_TYPE)</field>
+      <field type="float[30]" name="std">The estimator innovation standard deviation sqrt(diag(S))</field>
+    </message>
+    <message id="340" name="ESTIMATOR_INNOV_COV">
+      <description>An estimator agnostic innovation covariance message. Note: Very large, not meant for use over radio.</description>
+      <field type="uint64_t" name="time_usec" units="us">Timestamp (microseconds since system boot or since UNIX epoch)</field>
+      <field type="uint8_t" name="n">Number of innovations, max 21, limited by mavlink protocol</field>
+      <field type="uint8_t[21]" name="id" enum="MAV_FIELD">An array describing field type (see MAV_FIELD)</field>
+      <field type="uint8_t[21]" name="sensor" enum="MAV_SENSOR_TYPE">An array describing the sensor associated with the field (see MAV_SENSOR_TYPE)</field>
+      <field type="float[210]" name="cov">The estimator covariance matrix (upper triangle), horizontally stacked with lengths [n, n-1, n-2, ..., 1]</field>
```
Comment (tridge) [inline message_definitions/v1.0/common.xml]: an array of 210 floats??? really?

**Comment by priseborough** [comment]
A 21x21 covariance matrix would only be of use for detailed EKF diagnostics, and its not useful for us anyway because we have 23 states in EKF2 and 24 states in EKF3.

**Comment by amilcarlucas** [comment]
Good catch @tridge. I changed it to 21.
Is that better or do we need 276 for a 24x24 upper triangle matrix ?

**Comment by priseborough** [comment]
At the moment we have no method to populate it because we do not use a three parameter attitude representation inside the filter. EKF2 uses an attitude error vector  and EKF3 uses a quaternion.

**Comment by amilcarlucas** [comment]
Should I remove that message from our fork entirely ?

**Comment by tridge** [comment]
@amilcarlucas that commit was later reverted upstream:
https://github.com/mavlink/mavlink/commit/e24ebf5e6c98afbfa600307f17d9000fe0f7d1f3
so we should just drop it

**Comment by tridge** [comment]
@priseborough thanks for looking at this
@amilcarlucas we should check to see if we actually generate any of the msgs that use the float[21]. If we don't then merging it is harmless for now

**Comment by amilcarlucas** [comment]
rebased and added a couple of commits on top

**Comment by tridge** [comment]
this looks much better, thanks!
testing it against ArduPilot master now

**Comment by tridge** [comment]
looks good to me. Only functional change for us seems to be the extension to BATTERY_STATUS
@WickedShell can you check that?

**Pull request merged**